parser_library/src/parsing/parser_impl.cpp

```diff
@@ -543,24 +543,24 @@ void parser_impl::transform_imm_reg_operands(semantics::collector& col, id_index
     instruction_name = *instruction;
     std::vector<std::pair<size_t, size_t>> replaced;
     auto opernds = &col.current_operands().value;
-    auto mnem_tmp = context::instruction::mnemonic_codes.find(*instruction);
-    if (mnem_tmp != context::instruction::mnemonic_codes.end())
+    if (auto mnem_tmp = context::instruction::mnemonic_codes.find(*instruction);
+        mnem_tmp != context::instruction::mnemonic_codes.end())
     {
         instruction_name = context::instruction::mnemonic_codes.at(*instruction).instruction->instr_name;
         replaced = context::instruction::mnemonic_codes.at(*instruction).replaced;
     }
     int position = 0;
-    for (auto& operand : *opernds)
+    for (const auto& operand : *opernds)
     {
         // replaced operands are skipped being immediate values
-        if (replaced.size() != 0 && replaced.at(position).first == position)
+        if (!replaced.empty() && replaced.at(position).first == position)
```

Comment:
I think this is wrong... The `replaced` vector contains `index, value` pairs, so if the replacement occurs at position 3, it contains a single pair `3,value`, so `.at(3)` will throw.
Also there may be multiple replacements in a row, so this should probably be some kind of loop.

@michalbali256 - do you concur?